packages/teams-js/src/public/runtime.ts

```diff
@@ -121,19 +122,84 @@ export const teamsRuntimeConfig: IRuntime = {
       backStack: {},
       fullTrust: {},
     },
-    people: {},
     remoteCamera: {},
     sharing: {},
     teams: {
-      fullTrust: {
-        joinedTeams: {},
-      },
+      fullTrust: {},
     },
     teamsCore: {},
     video: {},
   },
 };
 
+// object of version constants
+export const versionConstants = {
+  '1.9.0': {
+    capabilities: { location: {} },
+    hostClientTypes: [
+      HostClientType.desktop,
+      HostClientType.web,
+      HostClientType.android,
+      HostClientType.ios,
+      HostClientType.rigel,
+      HostClientType.surfaceHub,
+      HostClientType.teamsRoomsWindows,
+      HostClientType.teamsRoomsAndroid,
+      HostClientType.teamsPhones,
+      HostClientType.teamsDisplays,
+    ],
+  },
+  '2.0.0': {
+    capabilities: { people: {} },
+    hostClientTypes: [
+      HostClientType.desktop,
+      HostClientType.web,
+      HostClientType.android,
+      HostClientType.ios,
+      HostClientType.rigel,
+      HostClientType.surfaceHub,
+      HostClientType.teamsRoomsWindows,
+      HostClientType.teamsRoomsAndroid,
+      HostClientType.teamsPhones,
+      HostClientType.teamsDisplays,
+    ],
+  },
+  '2.0.1': {
+    capabilities: { teams: { fullTrust: { joinedTeams: {} } } },
+    hostClientTypes: [
+      HostClientType.android,
+      HostClientType.teamsRoomsAndroid,
+      HostClientType.teamsPhones,
+      HostClientType.teamsDisplays,
+    ],
+  },
+};
+
+export function generateBackCompatRuntimeConfig(highestSupportedVersion: string): IRuntime {
+  let newSupports = { ...teamsRuntimeConfig.supports, ...{} };
```

Comment:
what does the `...{}` part do?

Reply:
This is the [Typescript spread syntax](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax) and allows us to "spread out" the `supports` content and create a new object free of modification restrictions. Without the spreading, there's a lot of Typescript unhappiness with adding onto the supports field.

Reply:
The `...{}` is an empty object spread out which just allows us to use the new object creator.

Reply:
I checked on this to see if we could get rid of it (It's what I saw all over stack overflow so I hadn't questioned it haha) and I think we can get rid of the ...{}. Thanks for pointing it out!

Reply:
It's the spreading of an empty object part that I was a bit curious about. If it's not needed let's remove it.

Reply:
Agreed, done!